llvmDebugInfoC/src/scripts/konan_lldb.py

```diff
@@ -284,10 +316,20 @@ def __init__(self, valobj):
         offset = zerro_address - valobj.unsigned
         size = first_address - zerro_address
         self._children = [MemberLayout(str(x), type, offset + x * size) for x in range(self.num_children())]
-        self._values = [self._read_value(i) for i in range(self._children_count)]
+        if VISUAL_DEBUGGER:
+            self._values = [None for index in range(self.cap_children_count())]
+        else:
+            self._values = [self._read_value(i) for i in range(self.cap_children_count())]
+
+    def cap_children_count(self):
+        #AppCode limits array len automatically
+        if CLIENT_XCODE:
+            return min(self._children_count, 20)
```

Comment:
Xcode will just show you all values in an array, so for example if you have 100k, you'll have problems. AppCode shows in increments of 50, which actually works well without modification.